### PR TITLE
Add vector store endpoints

### PR DESCRIPTION
**Pull request by willywg**
Hi...

I'm excited to submit this PR for the issue  #463: "Add the Vector Store, Vector Store Files, and Vector Store Batch Files APIs." With the recent release of OpenAI's Assistant V2, new endpoints have been introduced, including VectorStores, VectorStoreFiles, and VectorStoreBatchFiles. These endpoints are essential for leveraging the enhanced capabilities of the updated Assistant.

In this PR, I have implemented these new endpoints, ensuring compatibility and extending our API's functionality. Additionally, I have updated the README.md with usage examples to facilitate a better understanding for developers integrating these services.

Please find below the summary of changes:
- Added new API endpoints for VectorStores, VectorStoreFiles, and VectorStoreBatchFiles.
- Updated the README.md with clear and concise examples for each of the new endpoints.
- Ensured that all changes adhere to the existing code standards and practices.

Best regards.

## All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](../blob/main/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?

### CODE REVIEW

**Comment by bvsatyaram** [comment]
I have been working on this branch for a couple of days. It is stable and works as expected. I think it is good to go.

When can we get this merged?

File: README.md [hunk 1]
```diff
@@ -39,7 +39,12 @@ Stream text with GPT-4, transcribe and translate audio with Whisper, or create i
     - [Embeddings](#embeddings)
     - [Batches](#batches)
     - [Files](#files)
+      - [For fine-tunning purposes](#for-fine-tunning-purposes)
```
Comment (adamnaamani) [inline README.md:42]:
```suggestion
      - [For fine-tuning purposes](#for-fine-tuning-purposes)
```

File: README.md [hunk 2]
```diff
@@ -585,7 +590,7 @@ These files are in JSONL format, with each line representing the output or error
 If a request fails with a non-HTTP error, the error object will contain more information about the cause of the failure.
 
 ### Files
-
+#### For fine-tunning purposes
```
Comment (adamnaamani) [inline README.md:593]:
```suggestion
#### For fine-tuning purposes
```

**Comment by sumitngupta** [comment]
Thank you for the work on this PR! Getting our setup updated to utilize it.

**Comment by sumitngupta** [comment]
Curious if you all had any problems uploading straight `txt` files to a vector store? We're able to upload json files and we saw as part of the docs CSV's were not supported but we're getting a 
`status"=>"failed", "last_error"=>{"code"=>"unsupported_file", "message"=>"The file type is not supported."}`

error for plain utf-8 txt files. We're still working through this but wanted to float the question in case anyone else experienced this.

**Comment by alexrudall** [comment]
Released in [v7.1](https://github.com/alexrudall/ruby-openai/releases/tag/v7.1.0)

**Comment by alexrudall** [comment]
Thanks for this excellent PR @willywg !

**Pull request merged**